### PR TITLE
lestarch: updating .hpp and .md files for clean doxygen run

### PR DESCRIPTION
**Pull request by LeStarch**
| | |
|:---|:---|
|**_Originating Project/Creator_**| Infrastructure |
|**_Affected Component_**| Documentation |
|**_Affected Architectures(s)_**|  |
|**_Related Issue(s)_**|  |
|**_Has Unit Tests (y/n)_**|  |
|**_Builds Without Errors (y/n)_**|  |
|**_Unit Tests Pass (y/n)_**|  |
|**_Documentation Included (y/n)_**|  |

---
## Change Description

1. Cleans up all comments to improve autogenerated documentation
2. Updates and cleans base documentation
3. Sets up documentation to be auto-update on push to devel

## Rationale

No more out-of-date documentation.

### CODE REVIEW

**Comment by LeStarch** [comment]
Note: after merging this PR the autodocs action will start.  It may fail, as the `release/documentation` branch may need to be "bootstrapped" with the latest changes.  Afterwards, a re-bootstrap may be required if and when editing `docs/index.md`.  That is easy enough to do:

```
git fetch
git checkout release/documentation
git merge origin/devel
<fix merge conflicts>
git push --force release/documentation
```

Should be done by a maintainer.

File: .github/actions/entrypoint.sh
Comment (Joshua-Anderson) [inline .github/actions/entrypoint.sh:7]: Nitpick: If you're overriding the entrypoint in the autodocs workflow, do we also need this conditional to swap the action?

**Review by Joshua-Anderson — APPROVED** [review]
This is awesome, all the automation you added will be so nice.

File: .github/actions/entrypoint.sh
Comment (LeStarch) [reply .github/actions/entrypoint.sh:7]: Yes.  The entrypoint, although documented on actions, doesn't work(?)  so I have to fallback to this.  Regardless, this setup functions in my repo, and given the complexity and time invested in getting this far, I really don't want to change it.

**Comment by jsoref** [comment]
I _think_ that this led to:
https://github.com/nasa/fprime/runs/2897509672
> /github/workspace/docs/doxygen/generate_docs.bash: line 35: python: command not found

**Pull request merged**
